docs/how_to/deploy/adreno.rst

```diff
@@ -0,0 +1,259 @@
+on
+Android <https://tvm.apache.org/docs/how_to/deploy_models/deploy_model_on_android.html>`__
+
+**Prerequisites**: Android NDK, Android Debug Bridge (adb), OpenCL
+
+For us to begin with, Android NDK, Android Debug Bridge and OpenCL must
+be installed and Android part of TVM must be builded.
+
+**Android NDK installation**: https://developer.android.com/ndk
+
+**Android Debug Bridge installation**:
+https://developer.android.com/studio/command-line/adb
+
+**OpenCL installation**: https://github.com/KhronosGroup/OpenCL-SDK.git
+
+You can also build the android part of TVM via console. From the root
+folder of TVM:
+
+::
+
+   mkdir build_android
+   cd build_android
+   cmake .. -DUSE_OPENCL=path/to/OpenCL -DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK_HOME}/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_NATIVE_API_LEVEL=android-28 -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON -DANDROID_STL=c++_static -DUSE_CPP_RPC=ON
+   make -jN tvm_runtime tvm_rpc
+
+where **N** is the number of cores available on your *CPU*.
+
+At this stage you have builded TVM for Adreno.
+
+Build and deploy model for Adreno
+---------------------------------
+
+As the deployment pipelines on Android and on Adreno do not have many
+differences, in this section we will focus on one and only necessary
+aspect to compile and deploy models for Adreno and will take a look at
+the generation of kernels with and without textures. In addition, the
+possibility of choosing a different precision for model compilation will
+be considered.
+
+| |Android deployment pipeline|
+| Fig.2 Deployment pipeline on Adreno devices ### Adreno target
+  Normally, when compiling models on Android using OpenCL, the
+  corresponding target is used
+
+.. code:: python
+
+   target="opencl"
+
+Using Adreno, we want to get all the benefits of textures, so we have to
+use the following target to generate texture leveraging kernels
+
+.. code:: python
+
+   target="opencl -device=adreno"
+
+Let’s write simple model and take a look at generated kernels for these
+two targets
+
+.. code:: python
+
+   import tvm
+   from tvm import relay
+   import numpy as np
+
+   input_shape=(1, 56, 56, 32)
+   filter_shape=(3, 3, 32, 64)
+   filter = np.random.rand(*filter_shape)
+
+   dtype="float32"
+   input = tvm.relay.var("input", shape=input_shape, dtype=dtype)
+   weight = tvm.relay.var("weight", shape=filter_shape, dtype=dtype)
+   D = relay.nn.conv2d(input, weight, padding=(1, 1), data_layout="NHWC", kernel_layout="HWIO", out_dtype=dtype)
+
+   mod = relay.Function([input, weight], D)
+   params = {
+   "weight": tvm.nd.array(filter)
+   }
+
+Classical opencl target:
+
+.. code:: python
+
+   target="opencl"
+
+   with tvm.transform.PassContext(opt_level=3):
+   graph, lib, params = relay.build_module.build(mod, target, params=params)
+   print(lib.imported_modules[0].get_source())
+
+Notice the generated convolution kernel and the presence of pointers in
+the initialization of the function:
```

Comment:
`the presence of pointers` - What does it mean? I see the OpenCL code first time and I don't understand why it is important.